### PR TITLE
New CLI command: import reconciliation by id

### PR DESCRIPTION
**Pull request by AgustinSilverfin**
Add the option to import reconciliations with the ID instead of the handle
`silverfin import-reconciliation --id <id>`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -110,13 +110,24 @@ function storeImportedReconciliation(reconciliationText) {
 }
 
 async function importExistingReconciliationByHandle(firmId, handle) {
-  reconciliationText = await SF.findReconciliationText(firmId, handle);
+  let reconciliationText = await SF.findReconciliationText(firmId, handle);
```
Comment (Woetfin) [inline index.js:113]: Clearer to make this a `const` since the value is not being overwritten?

File: index.js [hunk 2]
```diff
@@ -110,13 +110,24 @@ function storeImportedReconciliation(reconciliationText) {
...
   if (!reconciliationText) {
     throw `${handle} wasn't found`;
   }
   storeImportedReconciliation(reconciliationText);
 }
 
+async function importExistingReconciliationById(firmId, reconciliationId) {
+  let reconciliationText = await SF.findReconciliationTextById(
```
Comment (Woetfin) [inline index.js:121]: Clearer to make this a `const` since the value is not being overwritten?

**Review by Woetfin — CHANGES_REQUESTED** [review]
See comments in code

File: api/sf_api.js [hunk 3]
```diff
@@ -232,6 +232,29 @@ async function findReconciliationText(firmId, handle, page = 1) {
...
+  // No data
+  if (reconciliations.length == 0) {
+    console.log(`Reconciliation ${reconciliationId} not found`);
+    return;
+  }
+  let reconciliationTexts = reconciliations.filter(
+    (element) => element["id"] === Number(reconciliationId)
+  );
+  if (reconciliationTexts.length != 0) {
+    for (let reconciliationText of reconciliationTexts) {
```
[the remark this answers is not in the document]
Comment (AgustinSilverfin) [reply api/sf_api.js:247]: Valid point, we still need to check if it has text (hidden templates). I'll adapt it

File: api/sf_api.js [hunk 4]
```diff
@@ -232,6 +232,25 @@ async function findReconciliationText(firmId, handle, page = 1) {
...
 
+async function findReconciliationTextById(firmId, reconciliationId, page = 1) {
+  const response = await fetchReconciliationTexts(firmId, page);
+  const reconciliations = response.data;
+  // No data
+  if (reconciliations.length == 0) {
+    console.log(`Reconciliation ${reconciliationId} not found`);
+    return;
+  }
+  let reconciliationText = reconciliations.filter(
```
Comment (Woetfin) [inline api/sf_api.js:243]: Should be a `const` too here

**Review by Woetfin — APPROVED** [review]
Small change from let to const, can still be made, but can be merged imo

**Pull request merged**